dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/utils/RuntimeUtilsTest.kt

```diff
@@ -1,31 +1,240 @@
+        devLogger.wtf(wtf)
+
+        assertThat(outputStream.toString())
+            .isEqualTo(
+                "D/Datadog: $debug\n" +
+                    "I/Datadog: $info\n" +
+                    "W/Datadog: $warning\n" +
+                    "E/Datadog: $error\n" +
+                    "A/Datadog: $wtf\n"
+            )
+        assertThat(errorStream.toString())
+            .isEmpty()
+    }
+
+    @Test
+    fun `devLogger should only print info or above`(
+        @SystemOutStream outputStream: ByteArrayOutputStream,
+        @SystemErrorStream errorStream: ByteArrayOutputStream
+    ) {
+        Datadog.setVerbosity(Log.INFO)
+
+        devLogger.v(verbose)
+        devLogger.d(debug)
+        devLogger.i(info)
+        devLogger.w(warning)
+        devLogger.e(error)
+        devLogger.wtf(wtf)
+
+        assertThat(outputStream.toString())
+            .isEqualTo(
+                "I/Datadog: $info\n" +
+                    "W/Datadog: $warning\n" +
+                    "E/Datadog: $error\n" +
+                    "A/Datadog: $wtf\n"
+            )
+        assertThat(errorStream.toString())
+            .isEmpty()
+    }
+
+    @Test
+    fun `devLogger should only print warning or above`(
+        @SystemOutStream outputStream: ByteArrayOutputStream,
+        @SystemErrorStream errorStream: ByteArrayOutputStream
+    ) {
+        Datadog.setVerbosity(Log.WARN)
+
+        devLogger.v(verbose)
+        devLogger.d(debug)
+        devLogger.i(info)
+        devLogger.w(warning)
+        devLogger.e(error)
+        devLogger.wtf(wtf)
+
+        assertThat(outputStream.toString())
+            .isEqualTo(
+                "W/Datadog: $warning\n" +
+                    "E/Datadog: $error\n" +
+                    "A/Datadog: $wtf\n"
+            )
+        assertThat(errorStream.toString())
+            .isEmpty()
+    }
+
+    @Test
+    fun `devLogger should only print error or above`(
+        @SystemOutStream outputStream: ByteArrayOutputStream,
+        @SystemErrorStream errorStream: ByteArrayOutputStream
+    ) {
+        Datadog.setVerbosity(Log.ERROR)
+
+        devLogger.v(verbose)
+        devLogger.d(debug)
+        devLogger.i(info)
+        devLogger.w(warning)
+        devLogger.e(error)
+        devLogger.wtf(wtf)
+
+        assertThat(outputStream.toString())
+            .isEqualTo(
+                "E/Datadog: $error\n" +
+                    "A/Datadog: $wtf\n"
+            )
+        assertThat(errorStream.toString())
+            .isEmpty()
+    }
+
+    @Test
+    fun `devLogger should only print assert or above`(
+        @SystemOutStream outputStream: ByteArrayOutputStream,
+        @SystemErrorStream errorStream: ByteArrayOutputStream
+    ) {
+        Datadog.setVerbosity(Log.ASSERT)
+
+        devLogger.v(verbose)
+        devLogger.d(debug)
+        devLogger.i(info)
+        devLogger.w(warning)
+        devLogger.e(error)
+        devLogger.wtf(wtf)
+
+        assertThat(outputStream.toString())
```

Comment:
A pleasure to see how Android makes mocking easy 💪, even deep to system streams.

Reply:
It's not that easy, but we developed a mechanism to actually mock that ourselves.